### DISCUSSION TITLE
Is there a way to highlight the search keyword in all results?

### DISCUSSION THREAD

**Discussion by tedtninety** [microsoft-search/pnp-modern-search#2746] [Q&A]
Hi again. I searched in numerous places before posting ( including here), but I can't find anything definitive on how to highlight the "keyword" in all results.. I'd imagine I would need to make  an entry into the "Edit Results Template" section, and I wouldn't know what that would be.

I checked out the "Edit Results Types", perhaps thinking that  that nay may be the key, however I am prompted "Data source field is required" and the drop downs are grayed out.

My goal is to have the keyword results hightlighted in yellow ( via CSS ). 

Any help would be greatly appreciated,.

**Comment by wobba** [comment]
By default the hithighlighted summary return should highlight the terms inputted. Here's a screenshot using the list template. If it's not highlighted that is an API issue, as the backend API's should provide a highlight. You should be able to test or repro using OOB search experiences in M365.

![image](https://user-images.githubusercontent.com/51104/217169141-98d906a5-10e9-4b12-886f-8ce5cf59641d.png)

**Comment by tedtninety** [comment]
Hi Mikael! Thank you so much for your reply. Apologies for not replying sooner, however I wanted to research and learn a bit more about the differences between "SharePoint Search" and "Microsoft Search", hoping that one or other may implement the hithighlights before replying.

However, I did learn that "Edit Result Types" appear for "Microsoft Search"; and "SharePoint Search" has it grayed out.
Also, that using multiple RefinableStrings for SearchTerms does not function  ( at least for me ) as used verbatim for "MS Search". I needed to add a path for it return results on "SharePoint" Search., but alas lose the ability to search through multiple fields.

Finally, sure enough, hithighlights does indeed function OOTB with search in SPO.
However, I have to failed to understand why it is not working for me for either of the aforementioned searches.

I hazard to ask: Under "Edit result types", is there a way that I may utilize it to "highlight" just the keyword via an inline template?

Thank you,
ted

**Reply by wobba** [reply]
Every result should have the input words highlighted if present, so you shouldn't have to do anything special. You can use the SharePoint query tool to test as well and you should see something like this:

![image](https://user-images.githubusercontent.com/51104/218839172-4bd7cb49-126f-4560-b8b2-b355bbf1185d.png)

**Comment by tedtninety** [comment]
OK thank you! I will have to go through my code and see what may be tripping it.

**Reply by nicolesherman00** [reply]
Curious if you figured out what was causing this for you? I am also having the same issue. Highlights will not show in my PNP webpart but OOTB it works

**Reply by jenniferpigeon** [reply]
Not yet, still troubleshooting. If I find a resolution I will be sure to add it to the post.

**Reply by tedtninety** [reply]
Hi, I'm sorry, I had not found a solution for this.

Scouring the threads , I keep seeing something about an API that some may or may not have working at their end.
I too have no issues with OOTB but using Custom Templates, the feature does not work.

To add, I never did find a solution to get "formatted" text to display for multiple lines text fields - which may also be that "API" issue.
